### DISCUSSION TITLE
One-Time Code Identity Validation: Displays Caps, Isn't Caps

### DISCUSSION THREAD

**Discussion by ghost** [authelia/authelia#6457] [Pre-Release Feedback]
When getting emailed a one-time code to verify one's identity, if you *type in* the code manually, the letters display as uppercase, but they are not, so when you try to verify, it just fails. If you hold shift or press capslock and type the letter as uppercase, the code will then work just fine.

There's no GUI indication of this either, btw, you just hit Verify and then it will spin for all eternity because the code failed.

authelia logs from a code failing:

> time="2023-12-25T09:46:50-06:00" level=error msg="Error occurred validating user session elevation One-Time Code challenge for user 'REDACTED': error occurred retrieving the code challenge from the storage backend" error="the code didn't match any recorded code challenges" method=PUT path=/api/user/session/elevation remote_ip=172.19.0.12 stack="github.com/authelia/authelia/v4/internal/handlers/handler_session_elevation.go:260 UserSessionElevationPUT\ngithub.com/authelia/authelia/v4/internal/middlewares/require_auth.go:19 Require1FA.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/bridge.go:54 handleRouter.(*BridgeBuilder).Build.func6.1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:56 SecurityHeadersCSPNone.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:82 SecurityHeadersNoStore.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/headers.go:12 SecurityHeaders.func1\ngithub.com/authelia/authelia/v4/internal/middlewares/delay.go:14 handleRouter.handleRouter.ArbitraryDelay.func11.func12\ngithub.com/fasthttp/router@v1.4.22/router.go:420 (*Router).Handler\ngithub.com/authelia/authelia/v4/internal/middlewares/log_request.go:14 handleRouter.LogRequest.func40\ngithub.com/valyala/fasthttp@v1.51.0/server.go:2359 (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.51.0/workerpool.go:224 (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.51.0/workerpool.go:196 (*workerPool).getCh.func1\nruntime/asm_amd64.s:1650 goexit"

**Comment by james-d-elliott** [comment]
They should be compared case insensitively, will double check this but it's possible the HMAC-SHA512 signature isn't generated correctly to perform the query.

**Reply by james-d-elliott** [reply]
Yeah that's all it was, next published manifest will fix it, should be about 5 mins. Will update the links after. Nice find!

**Reply by ghost** [reply]
Confirmed fixed.
